RegistrationApp/src/main/java/app/util/UserUtils.java

```diff
@@ -0,0 +1,206 @@
+                user.setDate(date);
+                check = false;
+            } else {
+                System.out.println("You entered wrong date, enter correct one again: ");
+            }
+        }
+    }
+
+    public void setUserEmail(BufferedReader bufferedReader, User user) throws IOException {
+        System.out.println("Enter email: ");
+        boolean check = true;
+        while (check) {
+            String email = bufferedReader.readLine();
+            if (validateEmail(email)) {
+                user.setEmail(email);
+                check = false;
+            } else {
+                System.out.println("You entered wrong email, enter correct one again: ");
+            }
+        }
+    }
+
+    public void setUserAvailable(BufferedReader bufferedReader, User user) throws IOException {
+        Random rd = new Random();
+        boolean isAvailable = rd.nextBoolean();
+        user.setAvailable(isAvailable);
+    }
+
+    public void setUserFirstName(BufferedReader bufferedReader, User user) throws IOException {
+        System.out.println("Enter Firstname: ");
+        String firstName = bufferedReader.readLine();
+        user.setFirstName(firstName);
+    }
+
+    public void setUserLastName(BufferedReader bufferedReader, User user) throws IOException {
+        System.out.println("Enter Lastname: ");
+        String lastName = bufferedReader.readLine();
+        user.setLastName(lastName);
+    }
+
+    public void setPhoneNumber(BufferedReader bufferedReader, User user) throws IOException {
+        System.out.println("Enter phone number: ");
+        boolean check = true;
+        while (check) {
+            String phoneNumber = bufferedReader.readLine();
+            if (validatePhoneNumber(phoneNumber)) {
+                user.setPhoneNumber(phoneNumber);
+                check = false;
+            } else {
+                System.out.println("You entered wrong phone number, enter correct one again: ");
+            }
+        }
+    }
+
+    public boolean validateUsername(String text) {
+        return text.matches("[a-zA-Z0-9]{4,}");
+    }
+
+    public  boolean validatePassword(String password) {
+        String pat = "(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9@#$%]).{4,}";
+        return (password.matches(pat));
+    }
+
+    public  boolean validateEmail(String email) {
+        String pattern = "\\w+@[a-zA-Z]+\\.[a-zA-Z]+";
+        return email.matches(pattern);
+    }
+
+    public boolean validateDate(String date) {
+        SimpleDateFormat df = new SimpleDateFormat("MM/dd/yyyy");
+        df.setLenient(false);
+        try {
+            df.parse(date);
+            return true;
+        } catch (ParseException ex) {
+            return false;
+        }
+    }
+
+    public boolean validatePhoneNumber(String number) {
+        return number.matches("^[+][0-9]{12}");
+    }
+
+    //task #1
+    public List<String> showUsersLogins(List<User> users) {
```

Comment:
apply correct format for streams to all methods

Reply:
fixed